File: PracticasProgramacion/Practica_25/cpp/main.cpp
/**
 * Practica_25 - Escribir y leer XML (datos basicos, arreglos, creados, ADT)
 */

#include "ManejadorXML.h"
#include <iostream>

int main() {
    ManejadorXML man;

    man.escribirDatosBasicos("datos_basicos.xml");
    man.escribirArreglo("arreglo.xml");
    man.escribirAutos("autos.xml");
    man.escribirPersonas("personas.xml");

    std::cout << "Leyendo datos basicos:\n";
    man.leerYMostrar("datos_basicos.xml");
    std::cout << "\nLeyendo arreglo:\n";
    man.leerYMostrar("arreglo.xml");
    std::cout << "\nLeyendo autos:\n";
    man.leerYMostrar("autos.xml");
    std::cout << "\nLeyendo personas (ADT):\n";
    man.leerYMostrar("personas.xml");

    return 0;
}

File: PracticasProgramacion/Practica_25/python/main.py
"""
Practica_25 - Escribir y leer XML (datos basicos, arreglos, autos, personas ADT).
"""

from manejador_xml import ManejadorXML

if __name__ == "__main__":
    man = ManejadorXML()
    man.escribir_datos_basicos("datos_basicos.xml")
    man.escribir_arreglo("arreglo.xml")
    man.escribir_autos("autos.xml")
    man.escribir_personas("personas.xml")

    print("Leyendo datos basicos:")
    man.leer_y_mostrar("datos_basicos.xml")
    print("\nLeyendo arreglo:")
    man.leer_y_mostrar("arreglo.xml")
    print("\nLeyendo autos:")
    man.leer_y_mostrar("autos.xml")
    print("\nLeyendo personas (ADT):")
    man.leer_y_mostrar("personas.xml")
